sys/progress_bar/progress_bar.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_progress_bar
+ * @{
+ *
+ * @file
+ * @brief       Progress bar implementation
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <assert.h>
+
+#include "progress_bar.h"
+
+void progress_bar_print(char *prefix, char *suffix, uint8_t value)
+{
+    if (value > 100) {
+        value = 100;
+    }
+
+    /* Hide cursor */
+    printf("\033[?25l");
+
+    /* Move cursor at the beginning of the line */
+    putchar('\r');
+
+    /* Display progress bar prefix if any */
+    if (prefix) {
+        printf("%s", prefix);
+    }
+
+    printf(PROGRESS_BAR_PREFIX_CHARACTER);
+
+    /* Fully reprint the progress bar */
+    for (unsigned i = 0; i < PROGRESS_BAR_LENGTH; i++) {
+        if (100 * i < (uint16_t)(value * PROGRESS_BAR_LENGTH)) {
+            printf(PROGRESS_BAR_FULL_CHARACTER);
+        }
+        else {
+            printf(PROGRESS_BAR_EMPTY_CHARACTER);
+        }
+    }
+
+    printf(PROGRESS_BAR_SUFFIX_CHARACTER);
+
+    /* Display progress bar suffix if any */
+    if (suffix) {
+        printf("%s", suffix);
+    }
+
+    /* show cursor */
+    printf("\033[?25h");
+
+    fflush(stdout);
+}
+
+void progress_bar_update(progress_bar_t *progress_bar)
+{
+    progress_bar_print(progress_bar->prefix, progress_bar->suffix,
+                       progress_bar->value);
+}
+
+void progress_bar_prepare_multi(uint8_t len)
+{
+    /* Give enough space to print all progress bars. */
+    for (uint8_t i = 0; i < len; ++i) {
```

Comment:
just seeing this: please be consistent with most of the codebase, use `i++` instead of `++i`.

Reply:
I think I already had this discussion some time ago and the conclusion was i++ or ++i in this case is exactly the same (see [here](https://stackoverflow.com/a/24858) for example). There's nothing about that in the [coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions). I personally prefer ++i in for loops.